### ISSUE TITLE
unexpected '-' when using --keep-git-dir

### ISSUE THREAD

**Issue by waterfoul** [hadolint/hadolint#951]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
Linting file passes

### Actual behavior
unexpected '-' expecting '\', at least one space, or only the --chown flag, the --chmod flag or the src and dest paths

### Steps to reproduce the behavior
Create a simple dockerfile with contents below run Hadolint

```bash
hadolint --version

Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:1.5-labs
FROM scratch as download
ADD --keep-git-dir https://github.com/hadolint/hadolint.git#v2.12.0 /hadolint
```

**Additional environment details (OS, stack version, etc.)**
I've tried with linux and macos. Looks to be a parsing issue. This is mentioned in https://github.com/hadolint/hadolint/issues/609 but they posted after that issue was merged so I'm opening a new one to get some visibility. This is something which is currently part of docker labs so Ideally it would still error with the ability to ignore it. Unfortunately DL1000 cannot be ignored. Since any custom syntax can be used to add more flags it might make sense to change the parser to parse all flags and pass them down to the linters instead of erroring on new flags

**Comment by tvd0x2a** [comment]
cross linking: https://github.com/hadolint/language-docker/pull/106

**Comment by m-ildefons** [comment]
Fixed in v2.15.0
